File: auth0/src/test/java/com/auth0/android/request/internal/CredentialsDeserializerMock.java
package com.auth0.android.request.internal;

import com.auth0.android.result.Credentials;
import com.auth0.android.result.CredentialsMock;

import java.util.Date;

class CredentialsDeserializerMock extends CredentialsDeserializer {

    @Override
    long getCurrentTimeInMillis() {
        return CredentialsMock.CURRENT_TIME_MS;
    }

    @Override
    Credentials createCredentials(String idToken, String accessToken, String type, String refreshToken, Date expiresAt, String scope) {
        return new CredentialsMock(idToken, accessToken, type, refreshToken, expiresAt, scope);
    }
}

File: auth0/src/test/java/com/auth0/android/request/internal/CredentialsDeserializerMock.kt
package com.auth0.android.request.internal

import com.auth0.android.result.Credentials
import com.auth0.android.result.CredentialsMock
import java.util.*

internal class CredentialsDeserializerMock : CredentialsDeserializer() {
    override fun createCredentials(
        idToken: String,
        accessToken: String,
        type: String,
        refreshToken: String?,
        expiresAt: Date,
        scope: String?,
        recoveryCode: String?
    ): Credentials {
        val credentials =
            CredentialsMock(idToken, accessToken, type, refreshToken, expiresAt, scope)
        credentials.recoveryCode = recoveryCode
        return credentials
    }

    override val currentTimeInMillis: Long
        get() = CredentialsMock.CURRENT_TIME_MS
}